### PR TITLE
lint: Disable `allowAny` lint rule

### PR DESCRIPTION
**Pull request by valarnin**
Followup to #4448.
Note that there are 5 lint errors in `util/index.ts` which are fixed as part of #4476, I didn't want to cause a merge conflict by fixing them here when basically that entire file got rewritten.

### CODE REVIEW

File: types/data.d.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface BaseOptions {
   GroupSpokenAlertsEnabled: boolean;
   Skin?: string;
   SystemInfo: SystemInfo;
+  Debug?: boolean;
```
[the remark this answers is not in the document]
Comment (valarnin) [reply types/data.d.ts:18]: Hmm, I'm not sure why that tsc error didn't happen locally, and why the instance in `oopsy_options.ts` wasn't showing up in my search in vscode. The eureka instance still holds, though.

File: .eslintrc.cjs [hunk 2]
```diff
@@ -256,7 +256,7 @@ const tsOverrides = {
     '@typescript-eslint/object-curly-spacing': ['warn', 'always'],
     '@typescript-eslint/strict-boolean-expressions': ['error', {
       // @TODO: Enable these keys over time
-      'allowAny': true,
+      'allowAny': false,
```
[the remark this answers is not in the document]
Comment (quisquous) [reply .eslintrc.cjs:259]: I would say just remove it entirely as we're working towards just using the defaults.

File: ui/oopsyraidsy/oopsy_options.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ const defaultOopsyNonConfigOptions: OopsyNonConfigOptions = {
 export type DeathReportSide = 'left' | 'right' | 'disabled';
 
 type OopsyConfigOptions = {
-  Debug: boolean;
+  Debug?: boolean;
```
Comment (quisquous) [inline ui/oopsyraidsy/oopsy_options.ts:50]: Can you make this non-optional, here and elsewhere? And then conditionals can just be `if (thing)` and not `if (thing === true)`.

File: ui/raidboss/popup-text.ts [hunk 4]
```diff
@@ -119,12 +119,10 @@ const triggerUpperCase = (str: string): string => {
...
 };
 
-// Disable no-explicit-any due to catch clauses requiring any
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const onTriggerException = (trigger: ProcessedTrigger, e: any) => {
+const onTriggerException = (trigger: ProcessedTrigger, e: unknown) => {
   // When a fight ends and there are open promises, from delaySeconds or promise itself,
   // all promises will be rejected.  In this case there is no error; simply return without logging.
-  if (!e)
+  if (e === null || typeof e !== 'object')
```
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/popup-text.ts:125]: Sounds good!

**Comment by valarnin** [comment]
Rebased to main (`e8s.ts` merge conflict) and made changes as requested.

**Pull request merged**